### PR TITLE
run update-cb3 on noarch if no host in requirements

### LINKED ISSUE #239: Noarch Python migrator change build to host
Currently the `noarch: python` migrator is adding reasonable things. Though it would be good if it could change `build` to `host` as well.

### LINKED ISSUE #240: Noarch python migrator should check for post-link/activate scripts
This is easy enough to do and should skip 60 jupyter-nbextension packages.

### PR DESCRIPTION
**Pull request by CJ-Wright**
closes #239
closes #240

### CODE REVIEW

**Comment by CJ-Wright** [comment]
don't merge yet need to add the #240 functionality

File: conda_forge_tick/migrators.xsh [hunk 1]
```diff
@@ -442,6 +442,8 @@ class Noarch(Migrator):
                         '  build:\n    - pip',
```
Comment (justcalamari) [inline conda_forge_tick/migrators.xsh]: This should be in host

File: conda_forge_tick/auto_tick.xsh [hunk 2]
```diff
@@ -66,8 +66,15 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
                                    fork=fork,
                                    gh=gh)
 
-    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
+    # if postscript/activate no noarch
+    if migrator.__class__.__name__ == 'Noarch' and not any(
+            x in os.listdir(recipe_dir) for x in [
+                '{}-post-link'.format(attrs['feedstock_name']),
+                'activate.sh']):
```
Comment (justcalamari) [inline conda_forge_tick/auto_tick.xsh]: What about pre-link and pre-unlink?

File: conda_forge_tick/auto_tick.xsh [hunk 3]
```diff
@@ -66,8 +66,15 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
                                    fork=fork,
                                    gh=gh)
 
-    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
+    # if postscript/activate no noarch
+    if migrator.__class__.__name__ == 'Noarch' and not any(
```
Comment (justcalamari) [inline conda_forge_tick/auto_tick.xsh]: Should this be `any` instead of `not any`?

File: conda_forge_tick/auto_tick.xsh (hunk 2 above)
Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.xsh]: idk @isuruf?

File: conda_forge_tick/migrators.xsh [hunk 4]
```diff
@@ -439,9 +439,14 @@ class Noarch(Migrator):
             if 'pip' not in attrs['req']:
                 replace_in_file(
                         '  build:',
-                        '  build:\n    - pip',
+                        '  host:\n    - pip',
                         'meta.yaml',
                         leading_whitespace=False)
+                replace_in_file(
+                    '  host:',
+                    '  host:\n    - pip',
```
Comment (justcalamari) [inline conda_forge_tick/migrators.xsh]: I think now pip will be there twice if it had build before.

File: conda_forge_tick/migrators.xsh (hunk 1 above)
Comment (justcalamari) [reply conda_forge_tick/migrators.xsh]: Build might not exist, but pip should still be added to host.

Comment (CJ-Wright) [reply conda_forge_tick/migrators.xsh]: ah ok I should reverse the setup that is there now.

File: conda_forge_tick/auto_tick.xsh [hunk 5]
```diff
@@ -66,8 +66,17 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
...
                                    gh=gh)
 
-    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
+    # if postscript/activate no noarch
+    if migrator.__class__.__name__ == 'Noarch' and any(
+            x in os.listdir(recipe_dir) for x in [
+                '{}-pre-unlink'.format(attrs['feedstock_name']),
+                '{}-pre-link'.format(attrs['feedstock_name']),
+                '{}-post-link'.format(attrs['feedstock_name']),
```
Comment (justcalamari) [inline conda_forge_tick/auto_tick.xsh]: Do these scripts always follow this naming convention? I think I've seen pre-link.sh/.bat before.

File: conda_forge_tick/auto_tick.xsh [hunk 6]
```diff
@@ -66,8 +66,20 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
                                    fork=fork,
                                    gh=gh)
 
-    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
+    # if postscript/activate no noarch
+    suffixs = ['pre-unlink', 'post-link', 'pre-link']
+    exts = ['.bat', '.sh']
+    no_noarch_files = [
+        '{}-{}.{}'.format(attrs['feedstock_name'], suffix, ext)
```
Comment (justcalamari) [inline conda_forge_tick/auto_tick.xsh]: I don't think they need the feedstock name. See https://github.com/conda-forge/widgetsnbextension-feedstock/tree/master/recipe.

**Comment by justcalamari** [comment]
Anything else @isuruf ?

File: conda_forge_tick/auto_tick.xsh [hunk 7]
```diff
@@ -66,8 +66,20 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
                                    fork=fork,
                                    gh=gh)
 
-    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
+    # if postscript/activate no noarch
+    suffixs = ['pre-unlink', 'post-link', 'pre-link']
```
Comment (isuruf) [inline conda_forge_tick/auto_tick.xsh]: You can add `activate` here

Comment (justcalamari) [reply conda_forge_tick/auto_tick.xsh]: And maybe change the name `suffixs` since they aren't suffixes now.

File: conda_forge_tick/migrators.xsh [hunk 8]
```diff
@@ -392,7 +392,6 @@ class Noarch(Migrator):
     sel_pat = re.compile('(.+?)\s*(#.*)?\[([^\[\]]+)\](?(2)[^\(\)]*)$')
     unallowed_reqs = ['toolchain', 'gcc', 'cython', 'clangdev']
     checklist = ['No compiled extensions',
-                 'No post-link or pre-link or pre-unlink scripts',
```
Comment (justcalamari) [inline conda_forge_tick/migrators.xsh]: I don't know if we should remove this, even if the bot does check. There might be some reason the bot missed the scripts, and we want maintainers to be sure the PR can be merged.

File: conda_forge_tick/auto_tick.xsh [hunk 9]
```diff
@@ -66,8 +66,19 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
...
                                    gh=gh)
 
-    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
+    # if postscript/activate no noarch
+    script_names = ['pre-unlink', 'post-link', 'pre-link', 'activate']
+    exts = ['.bat', '.sh']
+    no_noarch_files = [
+        '{}.{}'.format(script_name, ext)
+        for script_names in suffixs for ext in exts
```
Comment (justcalamari) [inline conda_forge_tick/auto_tick.xsh]: `script_name in script_names`

File: conda_forge_tick/auto_tick.xsh [hunk 10]
```diff
@@ -66,8 +66,19 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
...
-    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
+    # if postscript/activate no noarch
+    script_names = ['pre-unlink', 'post-link', 'pre-link', 'activate']
+    exts = ['.bat', '.sh']
+    no_noarch_files = [
+        '{}.{}'.format(script_name, ext)
+        for script_name in script_names for ext in exts
+        ]
+    if migrator.__class__.__name__ == 'Noarch' and any(
```
Comment (scopatz) [inline conda_forge_tick/auto_tick.xsh]: I think this should be `if isinstance(migrator, Noarch)`, rather than checking the classname

**Comment by CJ-Wright** [comment]
Thank you!

**Pull request merged**